### PR TITLE
[MRG+1] Fix Categorical dimension not supporting exactly 2 categories

### PR DESCRIPTION
**Pull request by nfcampos**

### CODE REVIEW

**Comment by MechCoder** [comment]
Thinking about it again, I'm fine with this behaviour.

File: skopt/tests/test_space.py [hunk 1]
```diff
@@ -84,6 +84,22 @@ def test_categorical_transform():
...
 
 
+def test_categorical_transform_binary():
+    categories = ["apple", "orange"]
+    cat = Categorical(categories)
+
+    apple = [0.]
+    orange = [1.]
+
+    assert_equal(cat.transform(["apple"]).size, cat.transformed_size)
```
Comment (MechCoder) [inline skopt/tests/test_space.py]: Can you just check this size explicilty, that is 

    `cat.transformed_size == 1`?

**Comment by MechCoder** [comment]
looks ok to me. I'll wait for @glouppe or @betatim to merge.

File: skopt/tests/test_space.py (hunk 1 above)
Comment (nfcampos) [inline skopt/tests/test_space.py]: why? it is exactly when `transformed_size` differs from `transform(...).size` that you get bugs when this is eg. inside `Space`, which is how we hit this bug...

Comment (MechCoder) [inline skopt/tests/test_space.py]: It is possible that `transformed_size` is the same as `transform(..).size` but both are wrong.

Comment (MechCoder) [inline skopt/tests/test_space.py]: I should worded it as "Can you also just check this size explicitly"

Comment (nfcampos) [inline skopt/tests/test_space.py]: ah, yes that makes sense. I've added it. went ahead and added the same size check for the non-binary case

**Comment by glouppe** [comment]
Thanks! Merging this

**Comment by betatim** [comment]
(For the historians: this supersedes #134)

**Pull request merged**